### ISSUE TITLE
Define _Services_ which are templates for a service that can be added to collections without disturbing the _Service_ definition.

### ISSUE THREAD

**Issue by sharpninja** [jenius-apps/nightingale-rest-api-client#52]
Templates would allow reusable services that can exist in multiple Collections.  _Services_ added to Collections would be __immutable__, meaning changes are only allowed in the _Service_ definition which ensures that tests run across Collections are always in sync.

**Comment by dpaulino** [comment]
Do you think there's some overlap with this feature request? https://github.com/jenius-apps/nightingale-rest-api-client/issues/10.

I'm asking because I think I may be able to kill two issues with one feature

**Comment by sharpninja** [comment]
Yes, essentially we need templates and that would be a good ticket to drive it.

**Comment by dpaulino** [comment]
Are there other http clients that have this feature that you're looking for? I think I mostly understand what you're saying, but having a visual example would also be helpful.

**Comment by sharpninja** [comment]
I would use JSON and allow nested values.
________________________________
